ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/clientlibs/editor/js/editDialog.js

```diff
@@ -155,14 +160,98 @@
         }
     }
 
+    /**
+     * hides the RTE/plain text field on the basis of rich text checkbox and
+     * resolves it's value.
+     *
+     * @param {HTMLElement} dialog The dialog on which the operation is to be performed.
+     * @param isTitleRichText is rich text checkbox selected.
+     * @param isToggled is this function being called on toggle of rich text checkbox.
+     */
+    function resolveRichText(dialog, isTitleRichText, isToggled) {
+        var title = dialog.find(BASE_TITLE)[0],
+            richTextTitle = dialog.find(BASE_RICH_TEXT_TITLE)[0],
+            richTextTitleDiv = dialog.find("[data-cq-richtext-editable='true'][data-wrapperclass='cmp-adaptiveform-base__richtexttitle']")[0];
+        if(isTitleRichText.checked){
+            hideGraniteComponent(title);
+            showGraniteComponent(richTextTitle);
+            copyTextValueToRte(title, richTextTitleDiv);
+        } else {
+            hideGraniteComponent(richTextTitle);
+            showGraniteComponent(title);
+            if(isToggled){
+                title.value = $('<div>').html(richTextTitleDiv.innerHTML).text();
+            }
+        }
+    }
+
+    /**
+     * hides the RTE/plain text field on the basis of rich text checkbox and
+     * resolves it's value for enumNAmes/Options.
+     *
+     * @param {HTMLElement} dialog The dialog on which the operation is to be performed.
+     * @param areOptionsRichText is rich text checkbox selected.
+     * @param isToggled is this function being called on toggle of rich text checkbox.
+     */
+    function resolveRichTextOptions(dialog, areOptionsRichText, isToggled) {
+        var enumNames = dialog.find(BASE_ENUMNAMES_VISIBLE),
+            richTextEnumNames = dialog.find(BASE_WRAPPER_RICH_TEXT_ENUMNAMES),
+            richTextEnumNamesDiv = dialog.find("[data-cq-richtext-editable='true'][data-wrapperclass='cmp-adaptiveform-base__richTextEnumNames']");
+        if(areOptionsRichText != null && areOptionsRichText.checked){
+            for (var i = 0; i < richTextEnumNames.length; i++) {
+                hideGraniteComponent(enumNames[i]);
+                showGraniteComponent(richTextEnumNames[i]);
+                copyTextValueToRte(enumNames[i], richTextEnumNamesDiv[i]);
+            }
+        } else {
+            for (var i = 0; i < richTextEnumNames.length; i++) {
+                hideGraniteComponent(richTextEnumNames[i]);
+                showGraniteComponent(enumNames[i]);
+                if(isToggled){
+                    enumNames[i].value = $('<div>').html(richTextEnumNamesDiv[i].innerHTML).text();
+                }
+            }
+        }
+    }
+
+    function copyTextValueToRte (textElem, richTextElem) {
+        richTextElem.innerHTML = Utils.encodeScriptableTags(textElem.value);
+    }
+
+    //Function to hide guide component
+    function hideGraniteComponent (elem) {
```

Comment:
Use the existing show/hide present in Utils

Reply:
done